### PR TITLE
Create a toolbar sample app to show how to use and customize the toolbar component.

### PR DESCRIPTION
**Pull request by pocmo**
This makes it easier to test and implement the configuration options we want to support in our toolbar. And it will be helpful to show case how to do certain things with the component.

In the sample app you can select from a list of sample toolbars and try them. For now this sample includes the default toolbar (no customization) and a configuration that resembles the focus toolbar (for phone and tablets) - as much as is supported for now.

### CODE REVIEW

File: samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        toolbar.addDisplayAction(reload)
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        // Create a menu that looks like the one in Firefox Focus
+        ////////////////////////////////////////////////////////////////////////////////////////////
+
+        val share = SimpleBrowserMenuItem("Share…") { /* Do nothing */ }
+        val homeScreen = SimpleBrowserMenuItem("Add to Home screen") { /* Do nothing */ }
+        val open = SimpleBrowserMenuItem("Open in…") { /* Do nothing */ }
+        val settings = SimpleBrowserMenuItem("Settings")  { /* Do nothing */ }
```
Comment (sputnikci) [inline samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt]: \[ktlint\] WARNING: \[no\-multi\-spaces\] Unnecessary space\(s\) in column 58

File: samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        // Create a menu that looks like the one in Firefox Focus
+        ////////////////////////////////////////////////////////////////////////////////////////////
+
+        val share = SimpleBrowserMenuItem("Share…") { /* Do nothing */ }
+        val homeScreen = SimpleBrowserMenuItem("Add to Home screen") { /* Do nothing */ }
+        val open = SimpleBrowserMenuItem("Open in…") { /* Do nothing */ }
+        val settings = SimpleBrowserMenuItem("Settings")  { /* Do nothing */ }
```
Comment (sputnikci) [inline samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt]: \[ktlint\] WARNING: \[no\-multi\-spaces\] Unnecessary space\(s\) in column 58

Comment (sputnikci) [inline samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt]: \[ktlint\] WARNING: \[no\-multi\-spaces\] Unnecessary space\(s\) in column 58

**Comment by pocmo** [comment]
Just pushed a bunch of more toolbar enhancement on top of this PR. Would like to sneak them into the 0.7 release. :)

**Pull request merged**